optaplanner-examples/src/main/java/org/optaplanner/examples/taskassigning_listvariable/swingui/TaskOverviewPanel.java

```diff
@@ -93,39 +95,39 @@ public void resetPanel(TaskAssigningSolution taskAssigningSolution) {
         removeAll();
         skillColorFactory = new TangoColorFactory();
         List<Employee> employeeList = taskAssigningSolution.getEmployeeList();
-        Map<Employee, Integer> employeeIndexMap = new HashMap<>(employeeList.size());
-        int employeeIndex = 0;
+        List<Task> unassignedTaskList = new ArrayList<>(taskAssigningSolution.getTaskList());
+        List<JButton> taskButtonList = new ArrayList<>(taskAssigningSolution.getTaskList().size());
+
+        int rowIndex = 0;
         for (Employee employee : employeeList) {
             JLabel employeeLabel = new JLabel(employee.getLabel(), new TaskOrEmployeeIcon(employee), SwingConstants.LEFT);
             employeeLabel.setOpaque(true);
             employeeLabel.setToolTipText(employee.getToolText());
-            employeeLabel.setLocation(0, HEADER_ROW_HEIGHT + employeeIndex * ROW_HEIGHT);
+            employeeLabel.setLocation(0, HEADER_ROW_HEIGHT + rowIndex * ROW_HEIGHT);
             employeeLabel.setSize(HEADER_COLUMN_WIDTH, ROW_HEIGHT);
             employeeLabel.setBorder(BorderFactory.createLineBorder(Color.BLACK, 1));
             add(employeeLabel);
-            employeeIndexMap.put(employee, employeeIndex);
-            employeeIndex++;
-        }
-        int panelWidth = HEADER_COLUMN_WIDTH;
-        int unassignedIndex = employeeList.size();
-        for (Task task : taskAssigningSolution.getTaskList()) {
-            JButton taskButton = createTaskButton(task);
-            int x;
-            int y;
-            if (task.getEmployee() != null) {
-                x = HEADER_COLUMN_WIDTH + task.getStartTime();
-                y = HEADER_ROW_HEIGHT + employeeIndexMap.get(task.getEmployee()) * ROW_HEIGHT;
-            } else {
-                x = HEADER_COLUMN_WIDTH + task.getReadyTime();
-                y = HEADER_ROW_HEIGHT + unassignedIndex * ROW_HEIGHT;
-                unassignedIndex++;
-            }
-            if (x + taskButton.getWidth() > panelWidth) {
-                panelWidth = x + taskButton.getWidth();
+
+            for (Task task : employee.getTasks()) {
+                taskButtonList.add(createTaskButton(task, rowIndex));
+                unassignedTaskList.remove(task);
             }
-            taskButton.setLocation(x, y);
-            add(taskButton);
+            rowIndex++;
         }
+        for (Task task : unassignedTaskList) {
+            taskButtonList.add(createTaskButton(task, rowIndex));
+            rowIndex++;
+        }
+
+        // Determine task table width...
+        int taskTableWidth = taskButtonList.stream()
+                .peek(this::add) // ...and add each task to the panel
+                .mapToInt(taskButton -> taskButton.getLocation().x + taskButton.getWidth())
```

Comment:
Using JButton.getLocation() is against Swing layout rules IIRC.

What happens if you resizes the window after loading a dataset? What after it starts consuming/producing? Does everything still work?